packages/admin-portal/src/components/pages/subscriptions/__tests__/index.test.tsx

```diff
@@ -0,0 +1,166 @@
+import Routes from '@/constants/routes'
+import appState from '@/reducers/__stubs__/app-state'
+import { DeveloperModelPagedResult } from '@reapit/foundations-ts-definitions'
+import { SubscriptionsFilterFormValues } from '@/components/ui/subscriptions/subscriptions-filter-form'
+
+const createStore = (loading: boolean, data?: DeveloperModelPagedResult) => {
+  return {
+    ...appState,
+    adminDevManagement: {
+      loading,
+      ...(data ? { data } : {}),
+    },
+  }
+}
+
+describe('AdminDevManagement', () => {
+  let store, mockStore
+  beforeEach(() => {
+    mockStore = configureStore()
+  })
+  it('should match a snapshot when LOADING false', () => {
+    store = mockStore(createStore(false, {}))
+    expect(
+      mount(
+        <ReactRedux.Provider store={store}>
+          <MemoryRouter initialEntries={[{ pathname: Routes.DEV_MANAGEMENT, key: 'adminDevManagementRoute' }]}>
+            <Subscriptions />
+          </MemoryRouter>
+        </ReactRedux.Provider>,
+      ),
+    ).toMatchSnapshot()
+  })
+
+  it('should show loader when LOADING true', () => {
+    store = mockStore(createStore(true, {}))
+    expect(
+      mount(
+        <ReactRedux.Provider store={store}>
+          <MemoryRouter initialEntries={[{ pathname: Routes.DEV_MANAGEMENT, key: 'adminDevManagementRoute' }]}>
+            <Subscriptions />
+          </MemoryRouter>
+        </ReactRedux.Provider>,
+      ),
+    ).toMatchSnapshot()
+  })
+
+  it('should render blank Info when data is empty', () => {
+    store = mockStore(createStore(true))
+    expect(
+      mount(
+        <ReactRedux.Provider store={store}>
+          <MemoryRouter initialEntries={[{ pathname: Routes.DEV_MANAGEMENT, key: 'adminDevManagementRoute' }]}>
+            <Subscriptions />
+          </MemoryRouter>
+        </ReactRedux.Provider>,
+      ),
+    ).toMatchSnapshot()
+  })
+})
+
+const mockRouterProps = getMockRouterProps({ page: 2 })
+
+describe('onPageChangeHandler', () => {
+  it('should return a function when executing', () => {
+    const onPageChangeHandlerFn = onPageChangeHandler(mockRouterProps.history, {
+      type: 'developerEdition',
+      developerId: '36bd47c8-b41c-4724-8848-12056856f344',
+    } as SubscriptionsFilterFormValues)
+    expect(onPageChangeHandlerFn).toBeDefined()
+
+    onPageChangeHandlerFn(2)
+    expect(mockRouterProps.history.push).toBeCalled()
+  })
+})
+
+describe('onSearchHandler', () => {
+  it('should return a function when executing', () => {
+    const onSearchHandlerFn = onSearchHandler(mockRouterProps.history)
+    expect(onSearchHandlerFn).toBeDefined()
+
+    onSearchHandlerFn(
+      {
+        type: 'developerEdition',
+        developerId: '36bd47c8-b41c-4724-8848-12056856f344',
+      } as SubscriptionsFilterFormValues,
+      {
+        setStatus: jest.fn(),
+      },
+    )
+    expect(mockRouterProps.history.push).toBeCalled()
+  })
+})
+
+// describe('closeDisableMemberModal', () => {
```

Comment:
I guess will implement when the disabled button works right?

Reply:
yup